oak_remote_attestation_amd/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+//! functionality as part of #1867.
+
+#![no_std]
+
+extern crate alloc;
+
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+use oak_remote_attestation::handshaker::{AttestationGenerator, AttestationVerifier};
+use serde::{Deserialize, Serialize};
+
+// TODO(#1867): Add remote attestation support and use real TEE reports.
+#[derive(Clone)]
+pub struct PlaceholderAmdAttestationGenerator;
+
+impl AttestationGenerator for PlaceholderAmdAttestationGenerator {
+    fn generate_attestation(&self, attested_data: &[u8]) -> anyhow::Result<Vec<u8>> {
+        PlaceholderAmdReport::new(attested_data)
+            .to_string()
+            .map(|s| s.as_bytes().to_vec())
+    }
+}
+
+// TODO(#1867): Add remote attestation support and use real TEE reports.
+#[derive(Clone)]
+pub struct PlaceholderAmdAttestationVerifier;
+
+impl AttestationVerifier for PlaceholderAmdAttestationVerifier {
+    fn verify_attestation(
+        &self,
+        attestation: &[u8],
+        expected_attested_data: &[u8],
+    ) -> anyhow::Result<()> {
+        let attestation_report = PlaceholderAmdReport::from_string(
+            core::str::from_utf8(attestation)
+                .map_err(|_err| anyhow::anyhow!("could not parse remote attestation report"))?,
+        )?;
+        let actual_attested_data = attestation_report.data;
+        if actual_attested_data == expected_attested_data {
+            Ok(())
+        } else {
+            Err(anyhow::anyhow!(
+                "invalid attested data; got: {:?}, expected: {:?}",
+                actual_attested_data,
+                expected_attested_data
+            ))
+        }
+    }
+}
+
+/// Placeholder implementation of TEE report for remote attestation.
+///
+/// <https://www.amd.com/system/files/TechDocs/56860.pdf#page=39>
+///
+/// TODO(#1867): Add remote attestation support and use real TEE reports.
+#[derive(Deserialize, Serialize, Debug, Default, PartialEq)]
+pub struct PlaceholderAmdReport {
+    /// Version number of this attestation report.
+    pub version: u8,
+    /// Security version number of SNP firmware.
+    pub svn: u8,
+    /// The installed version of firmware.
+    pub platform_version: u8,
+    /// Arbitrary data to put into the TEE report.
+    pub data: Vec<u8>,
+    /// TEE measurement, i.e. VM hash.
+    pub measurement: Vec<u8>,
+    /// Signature of this report.
+    pub signature: Vec<u8>,
+}
+
+// TODO(#1867): Add remote attestation support.
+const TEST_TEE_MEASUREMENT: &str = "Test TEE measurement";
+
+impl PlaceholderAmdReport {
+    /// Placeholder function for collecting TEE measurement of remotely attested TEEs.
+    pub fn new(data: &[u8]) -> Self {
+        Self {
+            measurement: TEST_TEE_MEASUREMENT.to_string().as_bytes().to_vec(),
+            data: data.to_vec(),
+            ..Default::default()
+        }
+    }
+}
+
+impl PlaceholderAmdReport {
+    pub fn from_string(input: &str) -> anyhow::Result<Self> {
+        serde_json::from_str(input)
```

Comment:
This can be fixed as part of #2842, but json is probably not a useful way to send the report to the client. For clients to verify the signature they would need to rebuild the exact bytes of the original report from the json encoding, so we could just send the bytes of the original report to save space and complexity.

Reply:
Yeah I think this JSON stuff is only here because of the x509 extension approach that @ipetr0v implemented a while ago. I agree we should not have JSON eventually, added a TODO.